test/envtest/k8s_objects_status_envtest_test.go

```diff
@@ -0,0 +1,198 @@
+			Namespace: gw.Namespace,
+			Name:      "backend-svc",
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name:     "http",
+					Protocol: corev1.ProtocolTCP,
+					Port:     80,
+				},
+			},
+		},
+	}
+	require.NoError(t, ctrlClient.Create(ctx, &backendService))
+	t.Cleanup(func() { _ = ctrlClient.Delete(ctx, &backendService) })
+
+	httpRoute := gatewayv1beta1.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: gw.Namespace,
+			Name:      uuid.NewString(),
+		},
+		Spec: gatewayv1beta1.HTTPRouteSpec{
+			CommonRouteSpec: gatewayv1beta1.CommonRouteSpec{
+				ParentRefs: []gatewayv1beta1.ParentReference{{
+					Name: gatewayv1beta1.ObjectName(gw.Name),
+				}},
+			},
+			Rules: []gatewayv1beta1.HTTPRouteRule{{
+				Matches: []gatewayv1beta1.HTTPRouteMatch{{
+					Path: &gatewayv1beta1.HTTPPathMatch{
+						Type:  lo.ToPtr(gatewayv1beta1.PathMatchPathPrefix),
+						Value: lo.ToPtr("/path"),
+					},
+				}},
+				BackendRefs: builder.NewHTTPBackendRef(backendService.Name).
+					WithPort(80).
+					ToSlice(),
+			}},
+		},
+	}
+	require.NoError(t, ctrlClient.Create(ctx, &httpRoute))
+	t.Cleanup(func() { _ = ctrlClient.Delete(ctx, &httpRoute) })
+
+	require.Eventually(t, func() bool {
+		nn := k8stypes.NamespacedName{Name: httpRoute.Name, Namespace: httpRoute.Namespace}
+		if err := ctrlClient.Get(ctx, nn, &httpRoute); err != nil {
+			t.Logf("failed to get httpRoute: %v", err)
+			return false
+		}
+		if len(httpRoute.Status.Parents) == 0 {
+			t.Logf("no gateway parent in httpRoute status")
+			return false
+		}
+		programmed, ok := lo.Find(httpRoute.Status.Parents[0].Conditions, func(c metav1.Condition) bool {
+			return c.Type == string(gatewayv1beta1.GatewayConditionProgrammed)
+		})
+		if !ok {
+			t.Logf("no programmed condition in httpRoute status")
+			return false
+		}
+		if programmed.Status != metav1.ConditionTrue {
+			t.Logf("programmed condition is not true")
+			return false
+		}
+		return true
+	}, time.Second*10, time.Millisecond*50)
+	t.Cleanup(func() { _ = ctrlClient.Delete(ctx, &httpRoute) })
+}
+
+// runManagerWithConfig runs the manager in a goroutine using configuration modified by modifyCfgFn.
+// It also sets up configuration parameters that are required for the Gateway API to work as expected.
+func runManagerWithConfig(ctx context.Context, t *testing.T, envcfg *rest.Config, gw gatewayv1beta1.Gateway, modifyCfgFn func(cfg *manager.Config)) {
+	cfg := ConfigForEnvConfig(t, envcfg)
+
+	cfg.PublishStatusAddress = []string{"127.0.0.1"}
+	cfg.PublishService = mo.Some(k8stypes.NamespacedName{
+		Name:      publishSvcName,
+		Namespace: gw.Namespace,
+	})
+	cfg.FeatureGates[featuregates.GatewayFeature] = true
+	cfg.FeatureGates[featuregates.GatewayAlphaFeature] = true
+
+	modifyCfgFn(&cfg)
+
+	go func() {
+		err := manager.Run(ctx, &cfg, util.ConfigDumpDiagnostic{}, logrus.New())
+		require.NoError(t, err)
+	}()
+}
+
+const publishSvcName = "publish-svc"
```

Comment:
Only used above, could potentially be moved inside the closure above?

Reply:
Actually, it should be used twice - I made a mistake and hardcoded it in the `deployGateway` function's publish service definition: c87f956